doc/developer/tracing.md

```diff
@@ -0,0 +1,231 @@
+# Introduction
+
+[`tracing`] is a set of crates used for _logging_ and _tracing_ of rust
+binaries. It is primarily composed of the [`tracing`] crate, which is the
+core way to interact with the framework, and the [`tracing-subscriber`] crate,
+which provides an API to compose systems that consume tracing data.
+
+
+## Logging
+
+The most basic way to use `tracing` is to use it like other logging crates like `log`,
+which additional semi-structured capabilities.
+
+
+Materialize binaries are setup such that [events] are printed on stderr.
+
+For example:
+```
+tracing::info!("processed {} items", items.len()")
+```
+produces:
+```
+2022-02-15T18:40:14.289898Z INFO my::module: processed 100 items
+```
+(See [this page for how the colors will look])
+
+You also also add semi-structured data to the event, for example:
+```
+tracing::debug!("processed {} items", items.len(), ?enum_value")
+```
+produces:
+```
+2022-02-15T18:40:14.289898Z DEBUG my::module: processed 100 items _enum_value_=EnumVariant
+```
+because `?field` means record `field` as its `Debug` impl. See [recording fields] for more
```

Comment:
Maybe use `?enum_value` and `enum_value` instead to be consistent with the example above.